### ISSUE TITLE
Fetch entire channel and threads

### ISSUE THREAD

**Issue by mwarkentin** [rneatherway/gh-slack#25]
This is a bigger request, but it would be nice if I could specify that I want to fetch an entire channel, including thread contents.

Use case: we have a 30 day retention policy on our Slack content, but would like to be able to archive our incident response channels for long-term. This tool is working great so far but it's missing out on the contents in the threads so I would need to go through and run it many times to capture each thread.

Seems valuable but also potentially complex (pagination (#4 ), how do you render threads mixed in with the output (`<details>` blocks could work quite well..) so I'm interested to hear if this is something you'd consider!

**Comment by rneatherway** [comment]
Right, interesting. Currently the tool wraps the whole output in `<details>` if you pass `-d`, which I like for keeping an archive neat as a reference in an issue comment.

I think you could more `<details>` tags to include threads collapsed by default, something like this:


<details>
<summary>Click to expand</summary>
<ul>
<li>  
<details>
  <summary>Click to expand</summary>
</details></li>
<li><details>
  <summary>Click to expand</summary>
</details></li>

</details>

It does get more complicated though because you start having to output more HTML.

If you want to go down this route I'd suggest teaching the existing `History` method (in the "Otherwise we read the general channel history") to fetch and output threads that it finds as well as a first step. The return type will need to change to allow nested responses (perhaps make it a struct and add a map from ids or timestamps to another HistoryResponse).

This is certainly doable, but it wouldn't be a five minute change! I am happy to answer questions if you do want to work on it.
